app/assets/javascripts/models/data/preferences.js.coffee

```diff
@@ -24,11 +25,16 @@ ns.Preferences = do (ko
         @fromJson(data)
         @isLoaded(true)
       else
-        getJSON '/users/site_preferences', (data, status, xhr) =>
-          console.log "Loaded site preferences, #{JSON.stringify(data)}"
-
-          @fromJson(data)
-          @isLoaded(true)
+        $.ajax '/users/site_preferences',
+          type: 'GET'
+          dataType: 'json',
+          async: false,
```

Comment:
I had a nice comment explaining this but then thought it was needed and then deleted it - lol.

The problem is there was a race condition when using the getJSON method.  Preferences, once instantiated, uses this AJAX call to in order to populate itself.  Therefore, if you instantiate it, and immediately check a preference - it won't be loaded yet and will return an incorrect value.  The asynchronous call to getJSON won't be done in time.

So, the answer is removing the asynchronous nature of the call.  Here I did this by switching getJSON to an .ajax call so as to have access to the async option.  It works and does fix the issue.  You're right that its been deprecated; but it hasn't been removed from our build of jQuery yet.  I doubt we'll be changing that jQuery version soon.  We can look at changing this do a different strategy for sure.